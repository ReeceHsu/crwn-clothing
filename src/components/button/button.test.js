import { render, screen } from '@testing-library/react';
import Button, { BUTTON_TYPE_CLASSES } from './button.component';

describe('button', () => {
	it('render button', () => {
		const { asFragment } = render(<Button />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should be disabled if isLoading is true', () => {
		render(<Button isLoading={true}>Test</Button>);
		expect(screen.getByRole('button')).toBeDisabled();
	});

	it('should render google button when passed google type', () => {
		render(<Button buttontype={BUTTON_TYPE_CLASSES.google}>Test</Button>);
		expect(screen.getByRole('button')).toHaveStyle('background-color: #4285f4');
	});

	it('should render inverted button when passed inverted type', () => {
		render(<Button buttontype={BUTTON_TYPE_CLASSES.inverted}>Test</Button>);
		expect(screen.getByRole('button')).toHaveStyle('background-color: white');
	});
});
