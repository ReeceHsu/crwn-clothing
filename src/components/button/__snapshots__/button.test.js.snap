// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render button 1`] = `
<DocumentFragment>
  <button
    class="sc-dkrFOg ddlRvK"
  />
</DocumentFragment>
`;
